provider/datagen/src/transform/cldr/displaynames/language.rs

```diff
@@ -46,22 +46,46 @@ impl IterableDataProvider<LanguageDisplayNamesV1Marker> for crate::DatagenProvid
     }
 }
 
-/// Substring used to denote alternative region names data variants for a given region. For example: "BA-alt-short", "TL-alt-variant".
-const ALT_SUBSTRING: &str = "-alt-";
+/// Substring used to denote short display names data variants for a given language. For example: "az-alt-short".
+const ALT_SHORT_SUBSTRING: &str = "-alt-short";
+/// Substring used to denote long display names data variants for a given language. For example: "az-alt-long".
+const ALT_LONG_SUBSTRING: &str = "-alt-long";
+/// Substring used to denote menu display names data variants for a given language. For example: "az-alt-menu".
+const ALT_MENU_SUBSTRING: &str = "-alt-menu";
```

Comment:
`suggestion`: Could you instead parse the region into `{ region: str, alt: Option<Alt> }` where `enum Alt { Short, Long, Menu }`? This will make the code easier to maintain in the long run.

Reply:
Discussed with @nordzilla @snktd @Manishearth. There's a certain simplicity to parsing the JSON in its CLDR schema, rather than re-mapping it into a more expressive structure. It's not clear that it makes the code more maintainable. Ultimately either approach is fine so I suggest we leave it up to @snktd.

Reply:
See the other way to do it here, in the Serde parser:

https://github.com/unicode-org/icu4x/blob/main/provider/datagen/src/transform/cldr/cldr_serde/numbers.rs

Reply:
Keeping this as is for now. If this gets complicated going forward, I will change it to what you and Zibi suggested.